### PR TITLE
Dwg

### PR DESCRIPTION
**Pull request by XueyangSong**

### CODE REVIEW

File: alg/DWG.m [hunk 1]
```diff
@@ -30,248 +30,384 @@
...
-    % print all valid indexes in this round
-    if params.logging
-        fprintf('valid indexes in round %d:\n', cnt);
-        for q = 1:length(valid_idx)
-            if valid_idx(q) == 1
-                fprintf('%d ', q);
-            end
+    %% begin selection process
+    % get valid x and y coordinates for the biggest gain
+    M = 0.0; % largest element
```
Comment (Orienfish) [inline alg/DWG.m]: These initialization are not used? Perhaps you can replace them with some comments, saying what does each of them represent.

File: alg/DWG.m [hunk 2]
```diff
@@ -30,248 +30,384 @@
...
-                end
+            % update
+            node1 = clusters(x, i);
+            node2 = clusters(y, i);
+            commMST(node1, node2) = dist(node1, node2);
+            commMST(node2, node1) = dist(node2, node1);
+        end
+    end
+    
+    %% update D, valid_idx, sq, and I matrices accordingly
```
Comment (Orienfish) [inline alg/DWG.m:181]: Could you spell out what does each of the symbols mean in these comments?
'Cause the readers cannot memorize what does each of them mean.

**Review by Orienfish — CHANGES_REQUESTED** [review]
The code is clean and looks good! I think your logic is correct, and I'm glad that you finally figured out the way to represent the clusters.
Several concerns:
1) I cannot interpret the result of DWG from the plot. Running the tutorial, I got a topology with a lot of cross-overs. Hope you could explain this on the meeting tomorrow.
2) A small suggestion on coding style. It is better to spell out the meaning of each variable, e.g. "update sensing quality matrix" rather than "update sq". For a reader, it's hard to memorize the meaning of all variables.
3) Also a suggestion on coding style. Currently the code is long. Is there some repeated processes that can be wrapped into a function and put at the end of the file? For example, for some update process (like sq, I), you can wrap them into functions. In this way, the major function will keep clean, which helps reader focus on the main flow and should be easier to understand.

**Comment by XueyangSong** [comment]
I have refactored the codes as required. 
Pushed to clear stash list for easier version control.
I'll add logging information as discussed in the meeting for my next commit.

**Pull request closed without merging**